controllers/util/util.go

```diff
@@ -103,3 +105,21 @@ func AssertEqual[T comparable](actual T, expected T, exitCode int) {
 		os.Exit(exitCode)
 	}
 }
+
+// IsAnnotationTruthy returns true if the annotation is present
+// and has a truthy value
+func IsAnnotationTruthy(obj metav1.Object, key string) bool {
+	annotations := obj.GetAnnotations()
+
+	if val, found := annotations[key]; found {
+		return strings.ToLower(val) == "true"
+	}
+	return false
+}
+
+// Execute the provided function if the condition is true.
+func If(cond bool, fn func()) {
+	if cond {
+		fn()
+	}
+}
```

Comment:
This util does not make sense. What would be a good reason to call a function and pass another function into it instead of just putting a normal if in place? 

Usually one creates an `IfElse` utils for cases where you want to treat the `if` as an expression (not as a statement).  In that case the implementation will be something like

```
func IfElse[T any](cond bool, trueVal, falseVal T) T {
    if cond {
       return trueVal
    } else {
        return falseVal
    }
}
```

To enable you doing stuff like:
```
instance := someStruct {
      ...
      b: util.IfElse(someCond, Val1, val2),
      ...
}
```

But this is not the case here and the implementation above does not provide any value. 
Please remove it

Reply:
> This util does not make sense.

Couldn't agree more :)